src/contents/en-gb/doc_dotnet.md

```diff
@@ -0,0 +1,326 @@
+
+**4. Authentication**
+
+Use `Authenticator.EncodeAuthRespAsync` API which use two parameters, AuthIpcReq and Response. Response is a boolean type representing whether you want to grant the permissions to an app or not. To grant the permissions use true, otherwise false. Once we get the response, decode it and register the app. Implementation code of the described process is as follows.
+```csharp
+var (_, reqMsg) = await Session.EncodeAuthReqAsync(authReq);
+var ipcReq = await authenticator.DecodeIpcMessageAsync(reqMsg);
+var authIpcReq = ipcReq as AuthIpcReq;
+var resMsg = await authenticator.EncodeAuthRespAsync(authIpcReq, true);
+var ipcResponse = await Session.DecodeIpcMessageAsync(resMsg);
+var authResponse = ipcResponse as AuthIpcMsg;
+var session = await Session.AppRegisteredAsync(appId, authResponse.AuthGranted);
+```
+
+
+***Using SAFE mock browser***
+
+**1. Download the SAFE mock browser** 
+
+Download the mock SAFE Browser as indicated by the file name `safe-browser-<version>-<platform>-dev.zip` (see pre-requisites for instructions)
+
+**2. Add mock flag**
+
+Add `SAFE_APP_MOCK` flag for your project in **Properties > Build > Conditional compilation symbols**.
+
+**3. Log in**
+
+Create an account on SAFE mock browser and log in (As this is  using mock any random arbitary string can be used as invite token).
+
+Keep the browser open and logged in to this account before proceeding with next steps.
+
+**4. Generate an `AuthReq`**
+
+An `AuthReq` instance is generated with the code below:
+```csharp
+var authReq = new AuthReq
+{
+    AppContainer = true,
+    App = new AppExchangeInfo 
+    { 
+        Id = "net.maidsafe.tutorial.dotnet", 
+        Scope = "", 
+        Name = "Hello SAFE Network", 
+        Vendor = "MaidSafe.net Ltd." 
+    },
+    Containers = new List<ContainerPermissions> 
+    { 
+        new ContainerPermissions 
+        { 
+            ContName = "_publicNames", Access = { Insert = true, Update = true, Delete = true} 
+        }
+    }
+};
+```
+*Note: Passing a list of `ContainerPermissions` allows the definition of various permissions provided to specific containers.*
+
+**5. Send `AuthReq` to the Authenticator**
+
+Once the `AuthReq` instance is initialised it is then used to generate the authorisation request and send it to the Authenticator using the following code:
+
+```csharp
+var encodedReq = await Session.EncodeAuthReqAsync(authReq);
+var url = $"safe-auth://{encodedReq}";
+System.Diagnostics.Process.Start(url);
+```
+**6. Grant access**
+
+Once the authorisation request is received the Authenticator launches and a pop-up dialogue prompts for access (Allow or Deny). The Authenticator then sends the response back to the application.
+
+Once the authorisation request is received, the Authenticator sends an authorisation response using the system URI mechanism. The system matches the URI scheme to the application and launches the application passing the URI as an argument.
```

Comment:
The beginning of this paragraph is almost identical to the previous paragraph which is a little confusing - would it make sense to add in the word "confirmation" to differentiate?
"Once the authorisation request confirmation is received..."

Reply:
Will update the doc once the content is finalized on the HackMD.